code/modules/antagonists/traitor/datum_traitor.dm

```diff
@@ -47,6 +50,7 @@
 			uplink.uplink_handler = uplink_handler
 		else
 			uplink_handler = uplink.uplink_handler
+		uplink_handler.traitor_datum = src
```

Comment:
I think I would prefer you to not do this and to instead hold a list of permanent objectives on the uplink handler instead, which is then linked to the objectives list.

It would be better to keep uplink handlers completely independent from the antag datum, and to let the antag datum handle the uplink handler instead.